lib/record_store/provider.rb

```diff
@@ -150,6 +154,12 @@ def retry_on_connection_errors(
         loop do
           begin
             return yield
+          rescue UnparseableBodyError
+            raise if max_retries <= 0
+            max_retries -= 1
+
+            waiter.message = "Waiting to retry since provider is unavailable"
```

Comment:
As I suggested in https://github.com/Shopify/record_store/pull/181#discussion_r492919001 you might want instead to update the class to support an optional `message` parameter when calling the `wait` method.

Here, that line changes the message which will then never be changed again depending on the error (we could alternate between the two kinds of errors for instance), while allowing a temporary parameter to override it "just this time" would make things better I feel!